mypy/util.py

```diff
@@ -134,3 +140,36 @@ def id(self, o: object) -> int:
             self.id_map[o] = self.next_id
             self.next_id += 1
         return self.id_map[o]
+
+
+if sys.version_info >= (3, 6):
+    PathType = Union[AnyStr, os.PathLike]
```

Comment:
Yes, it works until someone calls ``get_type_hints`` on this function. Generally, this is considered a temporary workaround, for example we prohibit things like ``'collections.Counter[str]'``, but here I think it is OK.

Also note, that when you write ``PathType = Union[AnyStr, os.PathLike[AnyStr]]``, you create a generic type alias, so that when you write just ``PathType``, it will be translated to ``Union[Any, os.PathLike[Any]]``. What you want is probably:
```python
def _replace(src: PathType[AnyStr], dest: PathType[AnyStr], timeout: float = 10) -> None: ...
```